docs: Include details on Notion CRM Auto-SSL column (#6908)
theopensystemslab/planx-new: doc/how-to/how-to-setup-custom-domains.md

Path: `validation-only` → `shared-final`

1. Update the [PlanX CRM on Notion](https://www.notion.so/opensystemslab/Plan-CRM-27c35d469ad1806c8f4dd95067ccf4ff).

>     Find the relevant council's page in our internal CRM, and set the 'Auto-SSL' column to "In progress"

Reviewer: just for consistency with prose style throughout :)

```suggestion
    Find the relevant council's page in our internal CRM, and set the 'Auto-SSL' column to "In progress".
```

>     > - Do steps 11-12 **now** and send the council both DNS records together.

Reviewer: ```suggestion
    > - Do steps 12-13 **now** and send the council both DNS records together.
```

>     > - Do step 13 **now** and bundle the application-level changes into your current PR (to be merged in step 9).

Reviewer: ```suggestion
    > - Do step 14 **now** and bundle the application-level changes into your current PR (to be merged in step 10).
```

  
6. **Wait** for the council to confirm they've added the record.

    If this is likely to take a while, open a PR with your initial commit and get it merged, so that the `customDomains.ts` source of truth is accurate on `main`.

    When it's done, you can then verify in the [AWS console](https://us-east-1.console.aws.amazon.com/acm/certificates/list) (make sure you're in the `us-east-1` region). The status of the specific domain on the mining cert should have changed from `Pending validation` to `Success`.

>     NB. The mining certs will 'fail' after 72hrs of attempting to validate their assigned domains, in which case you cannot use it to verify a correct DNS record. However, they can be re-created by simply deploying `certificates` again, as in step 2. Note that when these certs fail, Pulumi loses track of them, so it will not clean them up on next deploy (i.e. we should do that manually).

Reviewer: ```suggestion
    NB. The mining certs will 'fail' after 72hrs of attempting to validate their assigned domains, in which case you cannot use it to verify a correct DNS record. However, they can be re-created by simply deploying `certificates` again, as in step 3. Note that when these certs fail, Pulumi loses track of them, so it will not clean them up on next deploy (i.e. we should do that manually).
```

    Commit this change, open a PR if you haven't already, and get the PR approved.

> 8. Deploy `certificates` again (as in step 2).

Reviewer: ```suggestion
8. Deploy `certificates` again (as in step 3).
```

    This deploy removes the domain from the mining cert and adds it to the shared cert proper. It does this by creating a new cert, but we **retain the old cert**, because for now the shared CDN still relies on it.

9.  Before we proceed, we need to **verify the new certificate has issued**.

>     **This will fail** if the council we are onboarding has not yet added the DNS validation record (which we should have already verified in step 5) - or if _any other council_ already using the shared CDN has removed their validation record since the last time the shared cert was replaced.

Reviewer: ```suggestion
    **This will fail** if the council we are onboarding has not yet added the DNS validation record (which we should have already verified in step 6) - or if _any other council_ already using the shared CDN has removed their validation record since the last time the shared cert was replaced.
```

NB. No new councils will be onboarded in the legacy mode, so when we migrate the last council, we can revise our documentation (i.e. delete this section).

1. Update the [PlanX CRM on Notion](https://www.notion.so/opensystemslab/Plan-CRM-27c35d469ad1806c8f4dd95067ccf4ff).

>     Find the relevant council's page in our internal CRM, and set the 'Auto-SSL' column to "In progress"

Reviewer: ```suggestion
    Find the relevant council's page in our internal CRM, and set the 'Auto-SSL' column to "In progress".
```

> 2. Do steps 3-5 from [flow A](#a-onboarding-a-new-council) (i.e. send the DNS validation record to the council).

Reviewer: ```suggestion
2. Do steps 4-6 from [flow A](#a-onboarding-a-new-council) (i.e. send the DNS validation record to the council).
```

    > NB. **If this is not the first domain on the shared CDN** and you want to expedite:

>     > - Do steps 4-5 **now** and send the council both DNS records together (you still have to wait for them to add the records before proceeding to step 2).

Reviewer: ```suggestion
    > - Do steps 5-6 **now** and send the council both DNS records together (you still have to wait for them to add the records before proceeding to step 2).
```

    Commit this, open a PR and get it approved.

> 4. Do steps 7-10 from [flow A](#a-onboarding-a-new-council) (i.e. initialise a new shared cert and attach the shared CDN to it).

Reviewer: ```suggestion
4. Do steps 8-11 from [flow A](#a-onboarding-a-new-council) (i.e. initialise a new shared cert and attach the shared CDN to it).
```

    Once again, commit these changes and open a PR.

11. Deploy the `application` layer by merging said PR and rolling it out to prod.

    This deploy tears down the legacy CloudFront distribution, which is now redundant because we have re-routed traffic to the shared distribution by virtue of the council swapping out the DNS record, and us then moving the alias across. It will also destroy the legacy (imported) cert with which the distribution was associated.

>     NB. Application-level config (see step 13 in [flow A](#a-onboarding-a-new-council)) is already in place for legacy councils, so no changes are needed in that regard.

Reviewer: ```suggestion
    NB. Application-level config (see step 14 in [flow A](#a-onboarding-a-new-council)) is already in place for legacy councils, so no changes are needed in that regard.
```
